packages/app/src/components/chart-tile.tsx

```diff
@@ -10,38 +10,37 @@ import { Heading } from './typography';
 import { Markdown } from './markdown';
 import { MetadataProps } from './metadata';
 
-type ChartTileProps = {
+interface ChartTileProps {
   title: string;
   metadata?: MetadataProps;
   description?: string;
-
   timeframeInitialValue?: TimeframeOption;
   disableFullscreen?: boolean;
-} & (
-  | // Check if the children are a function to support the timeframe callback, otherwise accept a normal react node
-  {
-      timeframeOptions?: undefined;
-      children: ReactNode;
-    }
-  | {
-      timeframeOptions: TimeframeOption[];
-      children: (timeframe: TimeframeOption) => ReactNode;
-    }
-);
+  timeframeOptions?: TimeframeOption[];
+  onSelectTimeframe?: (timeframe: TimeframeOption) => void;
+  children: ReactNode;
+}
 
 export function ChartTile({
   title,
-  description,
-  children,
   metadata,
-  timeframeOptions,
+  description,
   timeframeInitialValue,
   disableFullscreen,
+  timeframeOptions,
+  onSelectTimeframe,
+  children,
 }: ChartTileProps) {
   const [timeframe, setTimeframe] = useState<TimeframeOption>(
     timeframeInitialValue || TimeframeOption.ALL
   );
 
+  useEffect(() => {
+    if (typeof onSelectTimeframe !== 'undefined') {
```

Comment:
```suggestion
    if (onSelectTimeframe) {
```
would also suffice I think :-)

Reply:
Agreed and changed.